frontend/crd/src/app/milestones-page/milestones/milestones.component.html

```diff
@@ -1,28 +1,49 @@
 
-<ul>
-  <li *ngFor="let milestone of milestones">
-    <div class="title-container" (click)="milestone.expanded = !milestone.expanded">
-      <span class="title">{{ milestone.name }}</span>
-      <span class ="arrow" [ngClass]="{ 'arrow-up': milestone.expanded, 'arrow-down': !milestone.expanded }"></span>
-    </div>
-    <ng-container *ngIf="milestone.expanded">
-      <div class="content">{{ milestone.yearLevel }}</div>
-      <ng-container *ngFor="let task of milestone.tasks">
-        <div class="content">
-          {{task.description}}
-        </div>
-      </ng-container>
 
-      <ng-container *ngFor="let event of milestone.events">
-        <div class="content">
-          <div>{{event.name}}</div>
-          <div >{{event.description}}</div>
-          <div >{{event.date | date:'short'}}</div>
-          <div >{{event.organizer}}</div>
-          <div >{{event.location}}</div>
-        </div>
-      </ng-container>
+<div class="year-container" *ngFor="let yearLevel of yearLevels">
+  <mat-card class="year-card">
+    <mat-card-header>
+      <div class="year-name">{{yearLevel}}</div>
+    </mat-card-header>
+  </mat-card>
 
+
+  <mat-accordion>
+    <ng-container *ngFor="let milestone of milestonesMap.get(yearLevel)">
+      <mat-expansion-panel>
+        <mat-expansion-panel-header>
+          <mat-panel-title>
+            <mat-checkbox [checked]="milestone.isComplete" color="primary" (click)="$event.stopPropagation()"></mat-checkbox>
+            {{milestone.name}}
+          </mat-panel-title>
+<!--  Use if we want to display a description -->
+<!--          <mat-panel-description>-->
+<!--            {{milestone.description}}-->
+<!--          </mat-panel-description>-->
+        </mat-expansion-panel-header>
+        <mat-expansion-panel class="task-expansion-panel" *ngFor="let task of milestone.tasks">
+          <mat-expansion-panel-header>
+            <div class="flex-vertical-align">
+              <mat-checkbox disabled [checked]="task.isComplete" color="primary" (click)="$event.stopPropagation()"></mat-checkbox>
+              <p>{{task.description}}</p>
```

Comment:
right now this shows up rather than the task name and even though it can expand, there's no additional information under the task. shouldn't there be some concise name and then this description shows up when the task is clicked the same as how the event description works?

Reply:
Yeah, I was thinking about that. The examples they gave us were super simple and just had one-line descriptions with no names. Should I add a name field to the tasks?

Reply:
i think so, mostly from a UI standpoint. "Meet with academic advisor to discuss class schedule and receive PIN for registration" is not the best thing to display in a list, especially when the event names are just a few words. rather than task name and task description, this is more like display name and full name